executor/index_lookup_merge_join.go

```diff
@@ -531,7 +537,7 @@ func (imw *innerMergeWorker) doMergeJoin(ctx context.Context, task *lookUpMergeJ
 			case <-ctx.Done():
 				return
 			}
-		} else {
+		} else if !imw.lookup.isClosed.Load().(bool) {
```

Comment:
It still has potentiality to panic,

1. execute line 540
2. `close(e.joinChkResourceCh[i])`
3. execute line 541